File: fe-pc/ui/src/main/java/com/example/ui/service/AuthService.java
package com.example.ui.service;

import com.example.ui.apis.AuthApi;
import com.example.ui.apis.UserApi;
import com.example.ui.config.ApiMethodsPrivate;
import com.example.ui.config.ApiMethodsPublic;
import com.google.gson.Gson;
import okhttp3.*;

import java.io.IOException;
import java.util.HashMap;
import java.util.Map;

import org.json.JSONObject;

public class AuthService {
    private static final OkHttpClient client = new OkHttpClient();

    public static String login(String email, String password) throws IOException {
        Gson gson = new Gson();

        // 🪄 Tạo một Map thay cho model
        Map<String, String> loginData = new HashMap<>();
        loginData.put("email", email);
        loginData.put("password", password);

        String json = gson.toJson(loginData); // giống như { email, password } trong JS

        RequestBody body = RequestBody.create(json, MediaType.parse("application/json"));

        Request request = new Request.Builder()
                .url(AuthApi.getLoginUrl())
                .post(body)
                .build();

        try (Response response = client.newCall(request).execute()) {
            if (!response.isSuccessful()) {
                throw new IOException("Unexpected code " + response);
            }

            return response.body().string();
        }
    }

    // update thông tin
    // public static JSONObject updateUserService(JSONObject body) {
    //     String url = UserApi.updateInforUserApi;
    //     return ApiMethodsPrivate.putRequest(url, body);
    // }


    // Hmaf Đăng ký
    public static JSONObject register(JSONObject body) throws IOException {
        String url = AuthApi.getRegisterUrl();
        return ApiMethodsPublic.postRequest(url, body);
    }
}

File: fe-mobile/lifetipsui/app/src/main/java/com/example/lifetipsui/service/AuthService.kt
package com.example.lifetipsui.service

import com.example.lifetipsui.apis.AuthApi
import com.example.lifetipsui.helper.ApiMethodsPublic
import kotlinx.serialization.json.JsonObject

class AuthService {
    companion object {
        suspend fun login(data: Map<String, String>): JsonObject? {
            return try {
                val response = ApiMethodsPublic.post<JsonObject>(AuthApi.loginApi, data)
                response
            } catch (e: Exception) {
                println(">> [AuthService] Error calling login API: ${e.localizedMessage}")
                e.printStackTrace()
                null
            }
        }

        suspend fun register(data: Map<String, String>): JsonObject? {
            return try {
                val response = ApiMethodsPublic.post<JsonObject>(AuthApi.registerApi, data)
                response
            } catch (e: Exception) {
                println(">> [AuthService] Error calling register API: ${e.localizedMessage}")
                e.printStackTrace()
                null
            }
        }
    }
}

